### PR TITLE
embed: stop *grpc.Server on *serveCtx serve error

### PR DESCRIPTION
**Pull request by gyuho**
I forgot to handle this in https://github.com/coreos/etcd/pull/8987.

If serve errors before *grpc.Server is sent to serversC,
it should be closed manually.

@xiang90

### CODE REVIEW

**Comment by xiang90** [comment]
can we add a test case to cover this?

**Comment by gyuho** [comment]
@xiang90 It's hard unless we refactor embed package to break it down to smaller pieces.

In integration tests, I tried triggering failure to this path by passing nil `*transport.TLSInfo` to serve, but it is being passed by value, so impossible to trigger error outside of `embed` package. 

https://github.com/coreos/etcd/blob/e3da56a8dfb88c7ee8f6c282c986fb5154700910/embed/etcd.go#L518

Another way that `*grpc.Server.Stop` could have been missed is when `sctx.registerGateway` fails, but this is also hard to test (`grpc.Dial` takes context, but this server side context is not exported).

File: embed/serve.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ func (sctx *serveCtx) serve(
 		opts := []grpc.DialOption{grpc.WithInsecure()}
 		gwmux, err := sctx.registerGateway(opts)
 		if err != nil {
+			gs.Stop()
```
Comment (xiang90) [inline embed/serve.go]: probably make error as named return. and we always try to stop gs if it is not nil

Comment (gyuho) [reply embed/serve.go]: Done.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**